src/cfnlint/__init__.py

```diff
@@ -344,6 +345,14 @@ def __init__(self, filename, template, regions=['us-east-1']):
             'Rules'
         ]
 
+    def __deepcopy__(self, memo):
```

Comment:
This code repeats four times -- is there a way to consolidate this into a base class or library?

Reply:
In some ways yes and others no.  We need to support deepcopy on custom objects so this one will have to remain.